### PR TITLE
Update to: - @mui/material: ^5.15.15 - @mui/styles: ^5.15.15"

### PR DESCRIPTION
**Pull request by nickgraz**
- @mui/material: ^5.15.15
- @mui/styles: ^5.15.15"

Add default empty check for empty strings to get rid of MUI warning for 'empty string not found in options'

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#513** <a href="https://app.graphite.dev/github/pr/terraware/web-components/513?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: package.json [hunk 1]
```diff
@@ -21,8 +21,8 @@
     "@emotion/styled": "^11.11.0",
     "@mui/icons-material": "^5.15.3",
     "@mui/lab": "^5.0.0-alpha.159",
-    "@mui/material": "^5.15.3",
-    "@mui/styles": "^5.15.3",
+    "@mui/material": "^5.15.15",
+    "@mui/styles": "^5.15.15",
```
Comment (nickgraz) [inline package.json:25]: @karthikbtf this seemed to do the trick. Was just a mismatch in the resolved ThemeProvider between the two packages.

File: src/components/Autocomplete/Autocomplete.tsx [hunk 2]
```diff
@@ -85,6 +87,9 @@ export default function Autocomplete({
 
   if (isEqual) {
     optionalArgs.isOptionEqualToValue = isEqual;
+  } else {
+    optionalArgs.isOptionEqualToValue = (option: unknown, value: unknown) =>
```
Comment (nickgraz) [inline src/components/Autocomplete/Autocomplete.tsx:91]: This ensures that if clients pass in an empty string as the selected value, which creates a warning, that it is still considered equal to `undefined`.

**Review by karthikbtf — APPROVED** [review]
Thanks 🙏

**Comment by karthikbtf** [comment]
@nickgraz sorry to bother again, wondering if this is failing in terraware-web now. @constanzauanini has run into similar errors in the past with mui/material upgrades.. not sure if she recalls how to fix it.

```
Compiled with problems:X

ERROR in src/App.tsx:29:36

TS2339: Property 'TwClrBaseGray025' does not exist on type 'Palette'.
    27 | const useStyles = makeStyles((theme: Theme) => ({
    28 |   container: {
  > 29 |     backgroundColor: theme.palette.TwClrBaseGray025,
       |                                    ^^^^^^^^^^^^^^^^
    30 |     backgroundImage:
    31 |       'linear-gradient(180deg,' +
    32 |       `${getRgbaFromHex(theme.palette.TwClrBaseGreen050 as string, 0)} 0%,` +


ERROR in src/App.tsx:32:39

TS2339: Property 'TwClrBaseGreen050' does not exist on type 'Palette'.
    30 |     backgroundImage:
    31 |       'linear-gradient(180deg,' +
  > 32 |       `${getRgbaFromHex(theme.palette.TwClrBaseGreen050 as string, 0)} 0%,` +
       |                                       ^^^^^^^^^^^^^^^^^
    33 |       `${getRgbaFromHex(theme.palette.TwClrBaseGreen050 as string, 0.4)} 100%)`,
    34 |     backgroundAttachment: 'fixed',
    35 |     minHeight: '100vh',


ERROR in src/App.tsx:33:39

TS2339: Property 'TwClrBaseGreen050' does not exist on type 'Palette'.
    31 |       'linear-gradient(180deg,' +
    32 |       `${getRgbaFromHex(theme.palette.TwClrBaseGreen050 as string, 0)} 0%,` +
  > 33 |       `${getRgbaFromHex(theme.palette.TwClrBaseGreen050 as string, 0.4)} 100%)`,
       |                                       ^^^^^^^^^^^^^^^^^
    34 |     backgroundAttachment: 'fixed',
    35 |     minHeight: '100vh',
    36 |     '& .navbar': {

```

**Comment by nickgraz** [comment]
👀 - probably need to update them both in `terraware-web`

**Comment by nickgraz** [comment]
Looks like this version didn't publish, probably the funky characters in the title. What a mess

**Comment by karthikbtf** [comment]
> Looks like this version didn't publish, probably the funky characters in the title. What a mess

Lolz yes I saw that with my revert as well. Wondering which other upgrade is breaking tw now. Let me check.

**Comment by karthikbtf** [comment]
> Looks like this version didn't publish, probably the funky characters in the title. What a mess

Would this fix it in publish-rc.yaml (single quotes)? not sure if that would still format the '%s'

```
    - name: Apply version bump (rc)
        run: npm version prerelease --preid=rc -m 'Publishing RC - %s'
```

**Comment by nickgraz** [comment]
It seems this is the line failing https://github.com/terraware/web-components/blob/main/.github/workflows/auto-version.yml#L41

It is strange that most of the double quotes get scrubbed, just not the last one.

**Comment by nickgraz** [comment]
It does seem to work if I change those to single quotes locally: 

```
npm version patch -m '%s - Update to: - @mui/material: ^5.15.15 - @mui/styles: ^5.15.15"'
v2.16.2
```

But then we potentially run into this causing issues if single quotes are in the title. I think it is OK to ignore and just not put any quotes in the title.

**Comment by karthikbtf** [comment]
> It does seem to work if I change those to single quotes locally:
> 
> ```
> npm version patch -m '%s - Update to: - @mui/material: ^5.15.15 - @mui/styles: ^5.15.15"'
> v2.16.2
> ```
> 
> But then we potentially run into this causing issues if single quotes are in the title. I think it is OK to ignore and just not put any quotes in the title.

Got it, yes ok to ignore.

**Pull request merged**